app/views/questions/new.html.erb

```diff
@@ -5,23 +5,74 @@
+              <button type="button" class="btn btn-primary btn-theme btn-generate-q" data-toggle="modal" data-target="#myModal">
+                Choose Qset
+              </button>
+              <span class="qset-display">
+                <% if !params[:qset].nil? %>
+                  <%= @qsets.find(params[:qset]).name %>
+                <% end %>
+              </span>
+              <!-- hidden input field to pass in qset_id into params on submit -->
+              <% if !params[:qset].nil? %>
+                <input type="text" name="question[qset_id]" id="qset_id_input" class="hide-display" value="<%= @qsets.find(params[:qset]).id %>">
+              <% else %>
+                <input type="text" name="question[qset_id]" id="qset_id_input" class="hide-display">
+              <% end %>
+              <!-- Modal -->
+              <div class="modal fade" id="myModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel">
+                <div class="modal-dialog" role="document">
+                  <div class="modal-content">
+                    <div class="modal-header">
+                      <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+                      <h4 class="modal-title" id="myModalLabel">Choose Qset</h4>
+                    </div>
+                    <div class="modal-body">
+                      <!-- displays qsets in hierarchical order -->
+                      <div class="list-group">
+                        <% @qsets.walk_tree do |qset, level| %>
+                          <% if level > 0 %>
+                            <% level_viz = '&nbsp;&nbsp;&nbsp;&nbsp;' * (level - 1) %>
+                            <% if qset.settings(:permissions).qset_type == 'subsets' %>
+                              <a href="#" class="list-group-item disabled">
+                                <%= level_viz.html_safe %>
+                                <span class="glyphicon glyphicon-folder-close"></span>
+                                <p class="qset-name-modal inline-p">
+                                  <%= '&nbsp;&nbsp;'.html_safe %>
+                                  <%= qset.name %>
+                                </p>
+                              </a>
+                            <% else %>
+                              <a href="#" class="list-group-item qset-selectable">
+                                <%= level_viz.html_safe %>
+                                <span class="glyphicon glyphicon-list-alt"></span>
+                                <p class="qset-name-modal inline-p">
+                                  <%= '&nbsp;&nbsp;'.html_safe %>
+                                  <%= qset.name %>
+                                </p>
+                                <p class="hide-display">
+                                  <%= qset.id %>
```

Comment:
If you want to associate the qset ID with this chunk of DOM, a standard way that doesn't involve creating and hiding new DOM elements is to use `data-*` attributes:

https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Using_data_attributes
https://api.jquery.com/data/

Then your JS would check the data properties of the element clicked, instead of querying the elements containing text (which is slower, and mixes the idea of information used solely for program logic vs user-facing information).